File: models/rooms.go
package models

type Room struct {
	ID          int     `json: "id`
	Owner       int     `json: "owner_id"`
	Location    string  `json:"location"`
	Price       float32 `json: "price"`
	Capacity    int     `json: "capacity"`
	Description string  `json:"description"`
}

File: frontend/js/rooms.js
import {fetchRooms} from "./api.js";

const roomsContainer = document.getElementById("rooms");

async function loadRooms(){
    try{
        const rooms = await fetchRooms();

        if(rooms.length == 0){
            roomsContainer.innerHTML = "<p>No rooms available. <p>";
            return;
        }

        roomsContainer.innerHTML = rooms
        .map(
            (room) => `
            <div class= "room-card">
            <h3> ${room.location}</h3>
            <p>  ${room.description}</p>
            <p> Price: ${room.price}</p>
            <p> Capacity: ${room.capacity}</p>
            </div>`

        )
        .join("");
    }catch (error){
        roomsContainer.innerHTML = "<p>Error loading rooms </p>";
    }
}

loadRooms();